osier/tech_library.py

```diff
@@ -0,0 +1,180 @@
+                            capital_cost=4924.6*(1/kW)*to_MDOLLARS,
+                            om_cost_variable=0.0*to_MDOLLARS,
+                            om_cost_fixed=58.24*(1/kW)*to_MDOLLARS,
+                            fuel_cost=36.6329*(1/(MW*hour))*to_MDOLLARS,
+                            ramp_up_rate=0.5,
+                            ramp_down_rate=0.5,
+                            lifecycle_co2_rate=3.7e-4*co2_eq_units,
+                            land_intensity=0.0,
+                            )
+battery = StorageTechnology(technology_name='Battery',
+                            capacity=0.81534126*GW,
+                            capital_cost=613*(1/kW)*to_MDOLLARS,
+                            om_cost_variable=0*to_MDOLLARS,
+                            om_cost_fixed=15.32*(1/kW)*to_MDOLLARS,
+                            fuel_cost=0*to_MDOLLARS,
+                            storage_duration=4,
+                            efficiency=0.85,
+                            initial_storage=0,
+                            capacity_credit=0.5,
+                            lifecycle_co2_rate=3.3e-5*co2_eq_units,
+                            land_intensity=0.0,
+                            )
+wind = Technology(technology_name='WindTurbine',
+                  renewable=True,
+                  dispatchable=False,
+                  fuel_type='wind',
+                  capacity=0*GW, 
+                  capital_cost=1180.6*(1/kW)*to_MDOLLARS,
+                  om_cost_fixed=33.11*(1/kW)*to_MDOLLARS,
+                  lifecycle_co2_rate=1.2e-5*co2_eq_units,
+                  land_intensity=0.0,
+                  capacity_credit=0.35)
+solar = Technology(technology_name='SolarPanel',
+                  renewable=True,
+                  dispatchable=False,
+                  fuel_type='solar',
+                  capacity=2.8103015*GW, 
+                  capital_cost=673.2*(1/kW)*to_MDOLLARS,
+                  om_cost_fixed=8.05*(1/kW)*to_MDOLLARS,
+                  lifecycle_co2_rate=3.7e-5*co2_eq_units,
+                  land_intensity=0.0,
+                  capacity_credit=0.19)
+
+
+def _get_names_technologies():
+    """
+    Returns a list of names and technologies in
+    the :mod:`osier.tech_library`.
+    """
+
+    current_module = sys.modules[__name__]
+    technology_list = []
+    name_list = []
+
+    for name, obj in inspect.getmembers(current_module):
+        if isinstance(obj, Technology):
+            technology_list.append(obj)
+            name_list.append(name)
+    
+    return name_list, technology_list
+
+
+def renewables_plus_storage():
+    """
+    Returns a list of technology objects including only
+    renewable technologies and storage options.
+    """
+
+    names_list, technology_list = _get_names_technologies()
+    technology_list = [tech
+                       for tech in technology_list 
+                       if ((tech.renewable) or (hasattr(tech, "storage_duration")))]
```

Comment:
Consider splitting this up for readability

Reply:
I'm not sure what you mean by readability? I find list comprehensions more readable than for-loops, and they're faster.

Reply:
Ah, I find for-loops more readable personally. But I was thinking having an external variable called `has_storage = hasattr(tech, "storage_duration")`, since having that all in the list comprehension seems like quite a bit. If you think it's good though you can leave it as is.